grails-app/services/org/pih/warehouse/core/OrganizationService.groovy

```diff
@@ -67,7 +67,7 @@ class OrganizationService {
             organization = new Organization()
             organization.name = name
             organization.partyType = PartyType.findByCode(Constants.DEFAULT_ORGANIZATION_CODE)
-            organization.code = code?:identifierService.generateOrganizationIdentifier(name)
+            organization.code = code?:organizationIdentifierService.generate(name)
```

Comment:
This does not have to happen now, but passing the organization object here (either in params or as an argument on its own) seems like the right approach.

Reply:
yeah I agree. Just didn't want to do the refactor at this point since it was already getting unwieldy.